main/tests/server/src/com/google/refine/expr/functions/arrays/UniquesTests.java

```diff
@@ -28,13 +28,47 @@
 
 import org.testng.annotations.Test;
 
+import com.google.refine.RefineTest;
+import com.google.refine.expr.ParsingException;
 import com.google.refine.util.TestUtils;
 
-public class UniquesTests {
+public class UniquesTests extends RefineTest {
     @Test
     public void serializeUniques() {
         String json = "{\"description\":\"Returns array a with duplicates removed\",\"params\":\"array a\",\"returns\":\"array\"}";
         TestUtils.isSerializedTo(new Uniques(), json);
     }
+
+    public void uniquesJsonArray() throws ParsingException {
+        String[] test = {"'[2,1,1,3]'.parseJson().uniques().toString()", "[2, 1, 3]"};
+        parseEval(bindings, test);
+        String[] test1 = {"'[2,2,null,null,3,3]'.parseJson().uniques().toString()", "[2, null, 3]"};
+        parseEval(bindings, test1);
+    }
+
+    @Test
+    public void uniquesArray() throws ParsingException {
+        String[] test = {"[2,1,1,3].uniques().toString()", "[2, 1, 3]"};
+        parseEval(bindings, test);
+        String[] test1 = {"[2,2,null,null,3,3,3].uniques().toString()", "[2, null, 3]"};
+        parseEval(bindings, test1);
+
+        String[] test2 = {"['z','b','c','c','a'].uniques().toString()", "[z, b, c, a]"};
+        parseEval(bindings, test2);
+        String[] test3 = {"['z','z',null,'c','a'].uniques().toString()", "[z, null, c, a]"};
+        parseEval(bindings, test3);
+
+        String[] test4 = {"[toDate(2020), '2018-03-02'.toDate(), toDate(2020)].uniques().toString()", "[2020-01-01T00:00Z, 2018-03-02T00:00Z]"};
+        parseEval(bindings, test4);
+    }
```

Comment:
Just in case, I think having a test5 for only nulls (thinking about that problem Owen had one time when updating the GREL null handling parts)
```suggestion
        String[] test5 = {"[null,null,null].uniques().toString()", "[null]"};
        parseEval(bindings, test5);
    }
```

Reply:
I don't think I understand that risk that @ostephens identified, but I've added the test.

As an aside, OpenRefine has a deeply held ambivalence about the equivalency of nulls & empty strings, so it doesn't surprise me that problems were encountered when trying to change null handling.

Reply:
@tfmorris Hi Tom, we changed that equivalency BECAUSE of the ambivalence documented in #820 and a few other issues, where a majority of the community wanted more precise handling and treating null/empty string separately, which I completely agreed with the majority of the community in that regard.  Hence, since Owen merged that work last year that all landed in [3.0 release](https://github.com/OpenRefine/OpenRefine/issues?q=is%3Aissue+milestone%3A3.0+is%3Aclosed) along with the Database Importer/Exporter which also has null handling (and the metadata fiasco you're now aware of), so we now treat `null` and empty strings entirely separately (well, I hope we treat them separately in all areas of OpenRefine, but that's probably not been 100% verified because of probably a few tests lacking in places)
Like in this issue #2265 on the SQL Exporter where @colognesdgs says when using the JSON importer, and then later using the SQL Exporter.

> The data is interpreted as null, but stored by openrefine as empty strings.

At any rate, you get the idea... we still have some soft bugs floating around regarding null/empty string I think in regards to that previous ambivalence, and so testing around `null` and empty string is very dear to my heart still to finally fully address especially for scientific data being ingested through JSON or SQL importers/exporters (part of the reason we were liked by CZI for the grant).